js/src/forum/addPostMentionPreviews.js

```diff
@@ -14,12 +14,12 @@ export default function addPostMentionPreviews() {
     const parentPost = this.attrs.post;
     const $parentPost = this.$();
 
-    this.$().on('click', '.UserMention, .PostMention', function (e) {
+    this.$().on('click', '.UserMention:not(.UserMention--deleted), .PostMention:not(.PostMention--deleted)', function (e) {
       m.route.set(this.getAttribute('href'));
       e.preventDefault();
     });
 
-    this.$('.PostMention').each(function() {
+    this.$('.PostMention:not(.PostMention--deleted)').each(function() {
```

Comment:
Maybe it'd be worth adding user previews just to explain what `[deleted]` and `[unknown]` actually mean to the user to prevent any confusion?

Reply:
sorry I'm not sure I understand what you mean by adding user previews, `PostMention--deleted` describes a post mention when the post was deleted, so it always contains `[unknown]`.

For post mentions `[deleted]` only shows when the post exists but the author is deleted, in which case the post preview does show, is that what you meant ?

Reply:
Sorry, yeah that's what I meant -- I haven't looked into the CSS classes for post-related things recently, my mistake. 🙈

Reply:
No worries, it does actually fall under this PR, that class is new and anything related to deleted user/post mentions is introduced here, so feel free to ask about/break anything!

Reply:
When you say previews, do you mean a tooltip? We don't currently show the user card on mentions, although I believe that's planned: https://github.com/flarum/core/issues/1018. A tooltip for deleted/unknown could be inconsistent.